src/solidClientHelpers/bookmarks.js

```diff
@@ -49,19 +52,43 @@ export const BOOKMARK_TYPE_IRI = "http://www.w3.org/2002/01/bookmark#Bookmark";
 export async function addBookmark(bookmarkIri, bookmarks, fetch) {
   const { dataset, iri } = bookmarks;
   const bookmarkTitle = getResourceName(bookmarkIri);
-  const bookmark = defineThing(
-    {},
-    (t) => addUrl(t, rdf.type, BOOKMARK_TYPE_IRI),
-    (t) => addStringNoLocale(t, dct.title, bookmarkTitle),
-    (t) => addUrl(t, RECALLS_PROPERTY_IRI, bookmarkIri),
-    (t) => addDatetime(t, dct.created, new Date())
+  const [existingBookmark] = getThingAll(dataset).filter(
+    (t) => getUrl(t, RECALLS_PROPERTY_IRI) === bookmarkIri
   );
+  let results;
+  if (existingBookmark) {
+    results = { response: bookmarks, error: null };
```

Comment:
Should we somehow notify the user that the bookmark already existed? Or is this a very edge-case?

Reply:
The problem is that if the bookmark already exists, what we should be doing is removing it, so this leaves things as they are until the user clicks again. This should only happen if the clicks are too close together anyway.